auth/realm/base/src/main/java/org/wildfly/security/auth/realm/FileSystemSecurityRealm.java

```diff
@@ -1177,22 +1371,220 @@ private void consumeContent(final XMLStreamReader reader) throws XMLStreamExcept
             }
         }
 
-        private boolean validNamespace(final String namespace) {
-            return ELYTRON_1_0.equals(namespace) || ELYTRON_1_0_1.equals(namespace);
+        private void verifyIntegrity() {
+            if (this.publicKey != null) {
+                boolean checksumValidity;
+                try {
+                    checksumValidity = validateChecksum(this.mainIndex, this.path, this.secretKey);
+                } catch (IllegalStateException | IOException | GeneralSecurityException e) {
+                    throw ElytronMessages.log.unableToAccessMainIndex(this.mainIndex.toString());
+                }
+                if (!checksumValidity) {
+                    if (!validateDigitalSignature(this.path.toString(), this.name)) {
```

Comment:
If the intention here is the only run the `validateDigitalSignature()` if the `checksumValidity()` is false, is this the correct way to go?

Reply:
@Ashpan sorry if I am asking something obvious. Can checksumValidity be false and validateDigitalSignature() return true? Is the integrity valid in such a case?

Reply:
Ah yes so it is possible. This would be the case if some attributes of the file changes but without really changing the contents of the file, like white spaces or something similar. That would make the checksumValidity false, which validateDigitalSignature remains true. Integrity is still valid in this scenario.